File: data/tutorial_wp/ex-2-times-2-answer.c
#include <limits.h>

/*@ 
  requires 0 <= x <= INT_MAX / 2;
  assigns  \nothing ;
  ensures  \result == 2 * x ;
*/
int times_2(int x){
  int r = 0 ;
  int i = 0 ;

  L: ;
  
  /*@
    loop invariant 0 <= x ;
    loop invariant r == 2 * i ;
    loop invariant i + x == \at(i + x, L) ;
    loop assigns r, x, i ;
    loop variant x ;
  */
  while(x > 0){
    r += 2 ;
    x -- ;
    i++ ;
  }
  return r;
}

File: transpiled_rust/tutorial_wp/ex-2-times-2-answer.rs
fn times_2(mut x: i32) -> i32 {
    let mut r: i32 = 0;
    let mut i: i32 = 0;

    while x > 0 {
        r += 2;
        x -= 1;
        i += 1;
    }
    return r;
}